src/lib/core.ts

```diff
@@ -38,3 +39,17 @@ Bool.fromBytes = function ([b]) {
   return Bool(!!b);
 };
 Bool.sizeInBytes = () => 1;
+
+Scalar.toFieldsCompressed = function (s: Scalar) {
+  let isConstant = s.toFields().every((x) => x.isConstant());
+  let constantValue: Uint8Array | undefined = (s as any).constantValue;
+  if (!isConstant || constantValue === undefined)
+    throw Error('Scalar.toFieldsCompressed is not available in provable code.');
```

Comment:
Nit: Could we communicate what "provable" code is better in this error message? For users that are brand new to the Mina dev experience, this can cause some confusion on what this really means. Could we simplify this as to provide a hint on the correct usage?